src/aria/popups/Popup.js

```diff
@@ -320,10 +320,19 @@ module.exports = Aria.classDefinition({
                  ariaUtilsDelegate.getMarkup(this._delegateId),
                  ' style="position:absolute;top:-15000px;left:-15000px;visibility:hidden;display:block;"'
             ];
-            var role = cfg.role;
-            if (cfg.waiAria && role) {
-                html.push(' role="' + role + '"');
+
+            if (cfg.waiAria) {
+                var role = cfg.role;
+                if (role) {
+                    html.push(' role="' + role + '"');
+                }
+
+                var labelId = cfg.labelId;
+                if (labelId) {
+                    html.push(' aria-labelledby="' + labelId + '"');
+                }
```

Comment:
Even if `role` and `labelId` are not supposed to contain quotes, I think it is good, to be sure we don't have injections, to either escape (with the utilities from `aria.utils.String`) `role` and `labelId` or to use `setAttribute` once the DOM element is available.